mongo/src/main/scala/it/agilelab/darwin/connector/mongo/MongoConnector.scala

```diff
@@ -1,19 +1,165 @@
+import org.mongodb.scala.connection.NettyStreamFactoryFactory
+import org.mongodb.scala.{Document, FindObservable, ListCollectionsObservable, MongoClient, MongoDatabase, Observable, SingleObservable, bson}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import collection.JavaConverters._
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+import org.bson.Document
+
+
+case class MongoConfig(user: String, password: String, hosts: List[String], db: String, table: String)
+
+class MongoConnector(config: Config) extends Connector with Logging {
+
+  require(config.hasPath(ConfigurationKeys.USER))
+  require(config.hasPath(ConfigurationKeys.DATABASE))
+  require(config.hasPath(ConfigurationKeys.PASSWORD))
+  require(config.hasPath(ConfigurationKeys.HOST))
+  require(config.hasPath(ConfigurationKeys.TABLE))
+
+  log.debug("Creating default MongoConfiguration")
+  val mongoConfig: MongoConfig = MongoConfig(
+    config.getString(ConfigurationKeys.USER),
+    config.getString(ConfigurationKeys.PASSWORD),
+    config.getStringList(ConfigurationKeys.HOST).asScala.toList,
+    config.getString(ConfigurationKeys.DATABASE),
+    config.getString(ConfigurationKeys.TABLE)
+  )
+  log.debug("Created default MongoConfiguration")
+
+  log.debug(s"Creating connections")
+  val mongoClient: MongoClient = createConnection(mongoConfig)
+  log.debug(s"Connection created")
+
+  //  In MongoDB tables are called collections.
+  //  Schema:
+  //    id:        idSchema         (long)
+  //    schema:    schema           (string)
+  //    name:      schema.name      (string)
+  //    namespace: schema.namespace (string)
+
+  /**
+   * return the MongoClient
+   * @param mongoConf : config with keys defined by ConfigurationKeys object
+   * @return MongoClient
+   */
+  private def createConnection(mongoConf: MongoConfig): MongoClient = {
+
+    val credential: MongoCredential = createCredential(mongoConf.user, mongoConf.db, mongoConf.password.toCharArray)
+
+    val hosts: List[ServerAddress] = mongoConf.hosts.map(host => new ServerAddress(host))
+
+    val settings: MongoClientSettings = MongoClientSettings.builder()
+      .credential(credential)
+      .applyToClusterSettings(builder => builder.hosts(hosts.asJava))
+      //.streamFactoryFactory(NettyStreamFactoryFactory())
+      .build()
+
+    MongoClient(settings)
+
+  }
+
+  private def parser: Parser = new Parser()
+
+  override def fullLoad(): Seq[(Long, Schema)] = {
+    log.debug(s"loading all schemas from table ${mongoConfig.table}")
+    val collectionNames =
+      mongoClient
+        .getDatabase(mongoConfig.db)
+        .listCollectionNames()
+
+    val schemas: Seq[(Long, Schema)] =
+      Await.result(
+        collectionNames.flatMap(collectionName =>
+          mongoClient.getDatabase(mongoConfig.db).getCollection(collectionName).find().map(document => {
+            val key = document.filterKeys(k => k == "id").head._2.asInt64().getValue
```

Comment:
Let's try to handle better the case where the document has a wrong "schema" (i.e. not directly throwing a Mongo exception, but a more specific exception).
I see at least two error cases:
1. the key is not there
2. the key is there but it's not of the correct type